## Analysis of Theme Satisfaction and Loyalty at Theme Park Touchpoints

In [1]:
import pandas as pd 
import numpy as np 

In [2]:
df = pd.read_csv("survey_responses.csv")
scale_qns = df.iloc[:, 14:27]
scale_qns.head()

14. Getting tickets was easy for me  \
0                                    6   
1                                    7   
2                                    5   
3                                    6   
4                                    6   

   15. Fees were fairly priced (F&B, admission tickets)  \
0                                                  4      
1                                                  5      
2                                                  3      
3                                                  1      
4                                                  7      

   16. I spent less time queueing than expected  \
0                                             4   
1                                             3   
2                                             3   
3                                             1   
4                                             7   

   17. Getting an express / fast pass is worth it  \
0                                               1   
1                                               2   
2                                               5   
3                                               1   
4                                               4   

   18. The park was clean and well maintained  \
0                                           7   
1                                           7   
2                                           7   
3                                           6   
4                                           6   

   19. There were adequate facilities provided (toilets, shelters, cooling or heating, etc.)  \
0                                                  7                                           
1                                                  6                                           
2                                                  5                                           
3                                                  6                                           
4                                                  6                                           

   20. I was able to navigate through the theme park easily  \
0                                                  7          
1                                                  7          
2                                                  6          
3                                                  5          
4                                                  6          

   21. The rides and attractions were enjoyable  \
0                                             6   
1                                             7   
2                                             5   
3                                             3   
4                                             5   

   22. The rides and attractions help reinforce the park’s theme and atmosphere  \
0                                                  7                              
1                                                  7                              
2                                                  5                              
3                                                  7                              
4                                                  7                              

   23. The rides and attractions are relevant today  \
0                                                 7   
1                                                 6   
2                                                 5   
3                                                 4   
4                                                 4   

   24. The park was adequately staffed  25. Staff were friendly and helpful  \
0                                    7                                    7   
1                                    7                                    6   
2                                    5                                    6   
3                                    4                           

In [6]:
summary = scale_qns.describe()
summary.loc[['mean', 'std']] = summary.loc[['mean', 'std']].round(0)
summary

14. Getting tickets was easy for me  \
count                                171.0   
mean                                   6.0   
std                                    1.0   
min                                    1.0   
25%                                    5.0   
50%                                    6.0   
75%                                    7.0   
max                                    7.0   

       15. Fees were fairly priced (F&B, admission tickets)  \
count                                              171.0      
mean                                                 4.0      
std                                                  2.0      
min                                                  1.0      
25%                                                  2.0      
50%                                                  4.0      
75%                                                  5.0      
max                                                  7.0      

       16. I spent less time queueing than expected  \
count                                         171.0   
mean                                            3.0   
std                                             2.0   
min                                             1.0   
25%                                             2.0   
50%                                             3.0   
75%                                             4.5   
max                                             7.0   

       17. Getting an express / fast pass is worth it  \
count                                           171.0   
mean                                              4.0   
std                                               2.0   
min                                               1.0   
25%                                               2.5   
50%                                               4.0   
75%                                               5.0   
max                                               7.0   

       18. The park was clean and well maintained  \
count                                       171.0   
mean                                          6.0   
std                                           1.0   
min                                           1.0   
25%                                           5.0   
50%                                           6.0   
75%                                           6.0   
max                                           7.0   

       19. There were adequate facilities provided (toilets, shelters, cooling or heating, etc.)  \
count                                              171.0                                           
mean                                                 5.0                                           
std                                                  1.0                                           
min                                                  1.0                                           
25%                                                  5.0                                           
50%                                                  6.0                                           
75%                                                  6.0                                           
max                                                  7.0                                           

       20. I was able to navigate through the theme park easily  \
count                                              171.0          
mean                                                 6.0          
std                                                  1.0          
min                                                  2.0          
25%                                                  5.0          
50%                                                  6.0          
75%                                                  6.0          
max                                                  7.0          

       21. The rides and attrac

### Net Promoter Score (Measuring Customer Loyalty)
Using a Net Promoter Score Metric, we define: <br>
Detractors (Score of 1-3 included): unlikely to recommend to others, could actively discourage future customers; Unhappy<br>
Passives (Score of 4-5):  tends not to actively promote the brand, but will not damage it either. <br>
Promoters (Score of 6-7): most enthusiastic and loyal customers, likely to act as brand ambassadors; Highly Satisfied


In [4]:
promoters = scale_qns.apply(lambda x: (x >= 6).sum())
passives = scale_qns.apply(lambda x: ((x >= 4) & (x <= 5)).sum()) 
detractors = scale_qns.apply(lambda x: (x <= 3).sum())

total = scale_qns.count()

promoter_pct = (promoters / total) * 100
detractor_pct = (detractors / total) * 100

nps_score = promoter_pct - detractor_pct
nps_score = nps_score.round(0)

print(nps_score)

overall_nps = nps_score.mean().round(0)
print(f"Overall NPS Score: {overall_nps}")

14. Getting tickets was easy for me                                                          58.0
15. Fees were fairly priced (F&B, admission tickets)                                        -39.0
16. I spent less time queueing than expected                                                -42.0
17. Getting an express / fast pass is worth it                                              -17.0
18. The park was clean and well maintained                                                   66.0
19. There were adequate facilities provided (toilets, shelters, cooling or heating, etc.)    51.0
20. I was able to navigate through the theme park easily                                     55.0
21. The rides and attractions were enjoyable                                                 56.0
22. The rides and attractions help reinforce the park’s theme and atmosphere                 56.0
23. The rides and attractions are relevant today                                             44.0
24. The park was ade

### Interpretation of Net Promoter Score (NPS)
NPS Score Range: -100 to 100

Positive NPS → More promoters than detractors → Good customer loyalty<br>
Negative NPS → More detractors than promoters → Needs improvement<br>
0-50 NPS → Neutral<br>
Above 50 → Excellent<br>

<br>

Using the Net Promoter Score Metric, we see that the overall score is Neutral (34). This suggests that most visitors are passive, and have limited customer loyalty, being unlikely to promote visiting Universal Studios to others.<br>
 Inspecting the NPS at individual touch points, we see that the points highlighted below have a negative NPS:

Fees were fairly priced (F&B, admission tickets)                                        -39.0<br>
I spent less time queueing than expected                                                -42.0<br>
Getting an express / fast pass is worth it                                              -17.0<br>

These points relate to wait times/queues (NPS of  -42), pricing (-39). This suggests a disatisfaction with these touchpoints, compromising customer loyalty and potential for customers to promote visiting USS to others. The low NPS (-17) relating to the value of getting a fastpass indicate that the current solution provided to wait-times is inadequate. Thus, these particular touchpoints lead to customers being "detractors", discouraging others visit the theme park.
<br>
<br>

One other touchpoint-related question recieved a neutral NPS:

The rides and attractions are relevant today                                             44.0

This question relates to the attractions as a touchpoint, more specifically, their relevancy. This neutral NPS suggests a mixed sentiment in customers, and implies that this touchpoint can be improved as well. <br>

Indeed, the question directly tounching on customer loyalty supports our findings:
I am likely to come back to this theme park again                                        43.0 <br>
With a neutral NPS on this question, we confirm that most customers are passives who are not likely to promote Universal Studios to others or visit frequently.

### Customer Satisfaction Score (CSAT)

For the CSAT, we define: <br>
Satisfied responses = Ratings 5, 6, or 7<br>
Neutral = Rating 4<br>
Dissatisfied = Ratings 1, 2, or 3

In [5]:
satisfied = scale_qns.apply(lambda x: (x >= 5).sum(), axis=0)  # Count 5-7 ratings

csat_score = (satisfied / total) * 100
csat_score = csat_score.round(0)

print(csat_score)

14. Getting tickets was easy for me                                                          80.0
15. Fees were fairly priced (F&B, admission tickets)                                         29.0
16. I spent less time queueing than expected                                                 25.0
17. Getting an express / fast pass is worth it                                               36.0
18. The park was clean and well maintained                                                   88.0
19. There were adequate facilities provided (toilets, shelters, cooling or heating, etc.)    81.0
20. I was able to navigate through the theme park easily                                     81.0
21. The rides and attractions were enjoyable                                                 84.0
22. The rides and attractions help reinforce the park’s theme and atmosphere                 84.0
23. The rides and attractions are relevant today                                             79.0
24. The park was ade

### Interpretation of CSAT
The above are expressed as percentages. <br>
Using CSAT as a metric provides a simplified and direct view of customer satisfaction.<br>
For the CSAT, we see that 2 touchpoints which have low satisfaction:
Fees(29%)<br>
Queue Times (25%) <br>

Additionally, the satisfaction for the current solution for long wait times is low, at 36%.

Hence, low satisfaction at these touchpoints corroborate our NPS metric.